packages/components/src/components/drawer/drawer.jsx

```diff
@@ -0,0 +1,87 @@
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+class Drawer extends React.PureComponent {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            is_open: true,
+        };
+    }
+
+    toggleDrawer = () => {
+        this.setState({ is_open: !this.state.is_open });
+        if (this.props.toggleDrawer) {
+            this.props.toggleDrawer(this.state.is_open);
+        }
+    };
+
+    render() {
+        const {
+            children,
+            className,
+            footer,
+            header,
+        } = this.props;
+
+        return (
+            <div className={classNames('dc-drawer',
```

Comment:
With `classNames` we can pass an object with classes that can be set conditionally. e.g.

```JavaScript
classNames('dc-drawer', {'dc-drawer--open', this.state.is_open });
```